# HG changeset patch
# User ptisnovs
# Date 1284983956 -7200
# Node ID bf87d5af43614d609a5251c43eea44c028500d02
# Parent  4d6c88bce53e4ec3902312abad5eaada33f77a23
6985992 Fix test/javax/swing/plaf/synth/Test6933784.java for OpenJDK 6
Summary: Testcase correction
Reviewed-by: darcy

diff -r 4d6c88bce53e -r bf87d5af4361 test/javax/swing/plaf/synth/Test6933784.java
--- openjdk/jdk/test/javax/swing/plaf/synth/Test6933784-old.java	Thu Jan 29 14:58:12 2009 +0300
+++ openjdk/jdk/test/javax/swing/plaf/synth/Test6933784.java	Mon Sep 20 13:59:16 2010 +0200
@@ -29,7 +29,7 @@
 */
 
 import javax.swing.*;
-import javax.swing.plaf.nimbus.NimbusLookAndFeel;
+import javax.swing.UIManager.LookAndFeelInfo;
 import javax.swing.plaf.synth.SynthLookAndFeel;
 import javax.swing.text.Element;
 import javax.swing.text.html.HTMLDocument;
@@ -43,11 +43,20 @@
 
         checkImages();
 
-        UIManager.setLookAndFeel(new NimbusLookAndFeel());
+        setNimbusLookAndFeel();
 
         checkImages();
     }
 
+    private static void setNimbusLookAndFeel() throws Exception {
+        for (LookAndFeelInfo info : UIManager.getInstalledLookAndFeels()) {
+            if ("Nimbus".equals(info.getName())) {
+                UIManager.setLookAndFeel(info.getClassName());
+                break;
+            }
+        }
+    }
+
     private static void checkImages() throws Exception {
         SwingUtilities.invokeAndWait(new Runnable() {
             public void run() {
